### PR TITLE
[tests] add traffic analysis for Cert_5_3_05_RoutingLinkQuality.py

### PR DESCRIPTION
**Pull request by hjian2017**

### CODE REVIEW

File: tests/scripts/thread-cert/command.py [hunk 1]
```diff
@@ -57,3 +57,23 @@ def check_link_accept(command_msg, destination_node):
...
+    # Verify ICMP Request is forwarding to the next node of the path.
+    for i in range(0, len_path):
+        node_msg = sniffer.get_messages_sent_by(path[i])
+        node_icmp_msg = node_msg.get_icmp_message(ipv6.ICMP_ECHO_REQUEST)
+
+        if i < len_path - 1:
+            next_node = nodes[path[i + 1]]
+            next_node_rloc16 = next_node.get_addr16()
+            assert next_node_rloc16 == node_icmp_msg.mac_header.dest_address.rloc, "Error: The path is unexpected."
+            temp = node_icmp_msg.icmp.header.type
```
Comment (librasungirl) [inline tests/scripts/thread-cert/command.py]: `temp` defined but not used

File: tests/scripts/thread-cert/command.py [hunk 2]
```diff
@@ -57,3 +57,23 @@ def check_link_accept(command_msg, destination_node):
...
+
+def check_icmp_path(sniffer, path, nodes):
+    """Verify message is forwarded along the path.
+    """
+    len_path = len(path)
+
+    # Verify ICMP Request is forwarding to the next node of the path.
+    for i in range(0, len_path):
+        node_msg = sniffer.get_messages_sent_by(path[i])
+        node_icmp_msg = node_msg.get_icmp_message(ipv6.ICMP_ECHO_REQUEST)
```
Comment (librasungirl) [inline tests/scripts/thread-cert/command.py]: could not find get_icmp_message()

File: tests/scripts/thread-cert/command.py [hunk 3]
```diff
@@ -57,3 +57,22 @@ def check_link_accept(command_msg, destination_node):
 
     destination_link_local = destination_node.get_ip6_address(config.ADDRESS_TYPE.LINK_LOCAL)
     assert ipv6.ip_address(destination_link_local) == command_msg.ipv6_packet.ipv6_header.destination_address, "Error: The destination is unexpected"
+
+def check_icmp_path(sniffer, path, nodes):
```
Comment (librasungirl) [inline tests/scripts/thread-cert/command.py]: rename to `check_ping_request_path`? as you are only checking `ICMP_ECHO_REQUEST` in the implementation.

Comment (hjian2017) [reply tests/scripts/thread-cert/command.py]: I will modify it to be able to check every type icmp message forward path.

File: tests/scripts/thread-cert/Cert_5_3_05_RoutingLinkQuality.py [hunk 4]
```diff
@@ -45,90 +47,93 @@ def setUp(self):
...
+        self.nodes[LEADER].add_whitelist(self.nodes[DUT_ROUTER1].get_addr64(), config.RSSI['LINK_QULITY_0'])
+        self.nodes[DUT_ROUTER1].add_whitelist(self.nodes[LEADER].get_addr64(), config.RSSI['LINK_QULITY_0'])
         time.sleep(70)
 
-        addrs = self.nodes[ROUTER3].get_addrs()
-        for addr in addrs:
-            if addr[0:4] != 'fe80':
-                self.assertTrue(self.nodes[LEADER].ping(addr))
-
+        # Verify the ICMPv6 Echo Request take the least cost path.
```
Comment (librasungirl) [inline tests/scripts/thread-cert/Cert_5_3_05_RoutingLinkQuality.py]: nit: take->took or takes ?
the least cost path -> the longer path ? (as said in Step 9).

**Comment by hjian2017** [comment]
@librasungirl

**Pull request merged**